tianshou/data/utils/segtree.py

```diff
@@ -41,27 +15,15 @@ class SegmentTree:
     2. Store the segment tree in a binary heap.
 
     :param int size: the size of segment tree.
-    :param str operation: the operation of segment tree. Choices are "sum",
-        "min" and "max". Default: "sum".
     """
 
-    def __init__(self, size: int,
-                 operation: str = 'sum') -> None:
+    def __init__(self, size: int) -> None:
         bound = 1
         while bound < size:
             bound *= 2
         self._size = size
         self._bound = bound
-        assert operation in ['sum', 'min', 'max'], \
-            f'Unknown operation {operation}.'
-        if operation == 'sum':
-            self._op, self._init_value = np.add, 0.
-        elif operation == 'min':
-            self._op, self._init_value = np.minimum, np.inf
-        else:
-            self._op, self._init_value = np.maximum, -np.inf
-        # assert isinstance(self._op, np.ufunc)
-        self._value = np.full([bound * 2], self._init_value)
+        self._value = np.zeros([bound * 2])
```

Comment:
It is great to have simplified this. Since your are reimplementing something existing, it is way better to specialize it as much as possible to fit your need while being as fast as possible.